README.md

```diff
@@ -1,4 +1,104 @@
 # KindedJ
 A collection of shared interfaces for evidence-based Higher Kinded Types in the JVM
 
-Gitter: https://gitter.im/KindedJ/Lobby
+[![Gitter chat](https://badges.gitter.im/KindedJ/KindedJ.png)](https://gitter.im/KindedJ/Lobby)
+
+## Projects that support KindedJ
+
+Name & Organisation - Language - Supported since version
+
+#### List
+
+*Add yourself here by [sending a Pull Request](https://github.com/KindedJ/KindedJ/compare)*
+
+[Kategory](https://github.com/kategory/kategory) - Kotlin - 0.3.6
+
+## Purpose
+
+A Higher Kinded Type represents a value that wraps another value. It would be equivalent to F\<A> if Java supported higher kinds. This allows library writers to create abstractions that work on any wrapper type in a completely generic fashion. You can see some examples in the projects supporting KindedJ listed above.
+
+The objective of KindedJ is to provide a common set of interfaces that enable cross-library and cross-language support for all opensource maintainers using evidence-based emulation in the JVM.
+
+## Usage
+
+Evidence-based Higher Kinded Types (HKTs) is a way of emulating HKTs even without explicit language support. 
+
+A *datatype* is any class who we want to make into an HKT. Examples include List, Future, Optional, Stream, Observable... To consider a class a datatype it has to implement the interface `Hk<F, A>`, or provide a wrapper class that does. In this interface, the generic parameter `F` is considered the *witness* and the generic parameter `A` the *value*.
+
+A *witness* type is an empty type without constructors. This type is unique globally and exclusively used to identify a single datatype when used as a generic parameter.
```

Comment:
We should consider adding here some references here to the original papers and info relevant to the emulation of higherkinds for lightweight polymorphism which is where this encoding is derived from.